recording/src/nextcloud/talk/recording/Participant.py

```diff
@@ -232,14 +245,54 @@ def startFirefox(self, width, height, env):
         options.add_argument(f'--width={width}')
         options.add_argument(f'--height={height}')
 
-        if disk_usage('/tmp').free < 134217728:
-            self._logger.warning('Less than 128 MiB available in "/tmp", strange failures may occur')
-
         service = FirefoxService(
             env=env,
         )
 
-        self.driver = webdriver.Firefox(
+        self.driver = FirefoxDriver(
+            options=options,
+            service=service,
+        )
+
+        self.bidiLogsHelper = BiDiLogsHelper(self.driver, self._parentLogger)
+
+    def startChromium(self, width, height, env):
+        """
+        Starts a Chromium instance.
+
+        :param width: the width of the browser window.
+        :param height: the height of the browser window.
+        :param env: the environment variables, including the display to start
+                    the browser in.
+        """
+
+        self.startCommon()
+
+        options = ChromeOptions()
+
+        options.binary_location = '/usr/bin/chromium-browser'
+
+        # "webSocketUrl" is needed for BiDi; this should be set already by
+        # default, but just in case.
+        options.set_capability('webSocketUrl', True)
+
+        options.add_argument('--use-fake-ui-for-media-stream')
+
+        # Allow to play media without user interaction.
+        options.add_argument('--autoplay-policy=no-user-gesture-required')
+
+        options.add_argument('--kiosk')
+        options.add_argument(f'--window-size={width},{height}')
+        options.add_argument('--disable-infobars')
+        options.add_argument('--disable-dev-shm-usage')
```

Comment:
Rather than unconditionally disabling `disable-dev-shm-usage` it would be good to do it only if [there is not enough space in `/dev/shm`](https://github.com/nextcloud/spreed/blob/eaee928da9aee60e567752908f3136b80b0157f3/docs/Talkbuchet-cli.py#L325-L335)

Reply:
The default 64 MiB is still not enough, at least in my testings. In general, I've been running into all sort of page load errors when using `shm`.

Around half a gibibyte seems to work in my test environment, but random crashes are still present.

If conditionally requiring a larger `shm` is acceptable (and consequently conditionally increasing the RAM requirement), which size should be hard-coded?

For reference:
- https://github.com/puppeteer/puppeteer/issues/1834
- https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md#tips
- https://github.com/c0b/chrome-in-docker/issues/1

Reply:
> The default 64 MiB is still not enough, at least in my testings. In general, I've been running into all sort of page load errors when using `shm`.
>
> Around half a gibibyte seems to work in my test environment, but random crashes are still present.

That is surprising. 64 MiB were set based on actual tests with Talkbuchet, and everything worked fine in that case. Maybe the difference is that rendering the remote videos for the recording server requires a lot more memory in `shm` (in the case of the load tests done by Talkbuchet the remote videos are not rendered, which could explain the lower memory usage). Or maybe the difference comes from the base Docker image used (Talkbuchet uses the Selenium image, which may have additional tweaks) :thinking:
 
> If conditionally requiring a larger `shm` is acceptable (and consequently conditionally increasing the RAM requirement), which size should be hard-coded?

Good question... I would like to do some tests first to try to find out what is going on here :-)

Also note that the reason not to unconditionally disable it is because using `shm`, at least in theory, should provide a better performance (and if I recall correctly that was mentioned directly by Chromium developers). Also I guess that if it is enabled by default there is a good reason for that :-)

Reply:
For now, I went ahead with 2 GiB. Source: https://github.com/SeleniumHQ/docker-selenium#--shm-size2g

Reply:
Unfortunately I will not have time to do the tests soon, and that is also [the value used in the script to run Talkbuchet](https://github.com/nextcloud/spreed/blob/52c717149b3a11a785d45d9066bbd1ea86628243/docs/Talkbuchet-run.sh#L214-L216), so fine by me :-)